server/src/main/java/org/elasticsearch/health/metadata/HealthMetadata.java

```diff
@@ -84,22 +97,95 @@ public Disk getDiskMetadata() {
         return diskMetadata;
     }
 
+    public ShardLimits getShardLimitsMetadata() {
+        return shardLimitsMetadata;
+    }
+
     @Override
     public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         HealthMetadata that = (HealthMetadata) o;
-        return Objects.equals(diskMetadata, that.diskMetadata);
+        return Objects.equals(diskMetadata, that.diskMetadata) && Objects.equals(shardLimitsMetadata, that.shardLimitsMetadata);
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(diskMetadata);
+        return Objects.hash(diskMetadata, shardLimitsMetadata);
     }
 
     @Override
     public String toString() {
-        return "HealthMetadata{diskMetadata=" + Strings.toString(diskMetadata) + '}';
+        return "HealthMetadata{diskMetadata=" + Strings.toString(diskMetadata) + ", shardLimitsMetadata=" + shardLimitsMetadata + "}";
+    }
+
+    /**
+     * Contains the thresholds needed to determine the health of a cluster from the amount of room available to create new shards. These
+     * values are determined by the elected master.
+     */
+    public record ShardLimits(int maxShardsPerNode, int maxShardsPerNodeFrozen) implements ToXContentFragment, Writeable {
+
+        public static final ShardLimits UNKNOWN = new ShardLimits(-1, -1);
```

Comment:
This special value means we now have to deal with 3 states:

- unknown
- actual values
- missing limits ( `null` )

Could the `unknown` special value be eliminated and treat `null` as unknown? 
This way we'll have to deal with 2 values logic.

Reply:
makes sense, will convert them to `null and actual values`